src/graphql.test.ts

```diff
@@ -9,3 +15,25 @@ test('exports supported GraphQL operation types', () => {
     'link',
   ])
 })
+
+test('pass proper type to request variable and data payload parameter for query', () => {
+  const op = graphql.query(GetUserDetailDocument, (req, res, ctx) => {
+    // If type doesn't match, this should fail compilation
+    const variables: GetUserDetailQueryVariables = req.variables
+
+    // MANUAL VERIFICATION: intellisense on `ctx.data` should be of type `DataContext<GetUserDetailQuery>`.
+    // Not sure how to test this.
+    ctx.data
+  })
+})
+
+test('pass proper type to request variable and data payload parameter for mutation', () => {
+  const op = graphql.mutation(LoginDocument, (req, res, ctx) => {
+    // If type doesn't match, this should fail compilation
+    const variables: LoginMutationVariables = req.variables
+
+    // MANUAL VERIFICATION: intellisense on `ctx.data` should be of type `DataContext<LoginMutation>`.
+    // Not sure how to test this.
+    ctx.data
```

Comment:
Manual verification: ![image](https://user-images.githubusercontent.com/720635/121344614-948aed80-c8e9-11eb-9e98-aff9e0db0d00.png)

Reply:
We should move this into the `test/typings/graphql.test-d.ts`, eventually.